source/Dig/base.util.xmlhelper.mxml.bmx

```diff
@@ -303,190 +279,140 @@ Type TXmlHelper
 	'  ["other"] or ["*"]
 	'  <obj><other><FIELDNAME>bla</FIELDNAME></other></obj>
 	Function FindValue:String(node:TxmlNode, fieldName:String, defaultValue:String, logString:String="", searchInChildNodeNames:String[] = Null, searchInChildNodeAttributes:Int = False, depth:Int = 0)
+		Local exists:Int
+		Return FindValue(node, fieldName, defaultValue, exists, logString, searchInChildNodeNames, searchInChildNodeAttributes, depth)
+	End function
+	
+
+	Function FindValue:String(node:TxmlNode, fieldName:String, defaultValue:String, valueExists:Int var, logString:String="", searchInChildNodeNames:String[] = Null, searchInChildNodeAttributes:Int = False, depth:Int = 0)
 		If node
 			'loop through all potential fieldnames ("frames|f" -> "frames", "f")
-			Local fieldNames:String[] = fieldName.ToLower().Split("|")
-
-			For Local name:String = EachIn fieldNames
-				'given node has attribute (<episode number="1">)
-				If depth = 0 or searchInChildNodeAttributes
-					Local attributeExists:Int
-					Local result:String = GetAttribute(node, name, attributeExists)
-					if attributeExists then Return result
-				EndIf
+			If fieldName.Find("|") > 0
+				Local fieldNames:String[] = fieldName.ToLower().Split("|")
 
-				For Local subNode:TxmlNode = EachIn GetNodeChildElements(node)
-					Local subNodeName:String = subNode.GetName()
-					If name.length = subNodeName.length and name = subNodeName.ToLower() Then Return subNode.GetContent()
-					If dataLS.EqualsLower(subNodeName) 
-						Local attributeExists:Int
-						Local result:String = GetAttribute(subNode, name, attributeExists)
-						If attributeExists Then Return result
-					EndIf
-					If searchInChildNodeNames And searchInChildNodeNames.length > 0
-						If searchInChildNodeNames[0] = "*" Or StringHelper.InArray(subNode.getName(), searchInChildNodeNames, False)
-							Return FindValue(subNode, fieldName, defaultValue, logString, searchInChildNodeNames, searchInChildNodeAttributes, depth + 1)
-						EndIf
-					EndIf
+				For Local name:String = EachIn fieldNames
```

Comment:
Ich denke, dieses Pattern könnte für Performanceverluste verantwortlich sein.
Hier wird für jeden Feldnamen neu gesucht. Es gibt aber zahlreiche Fälle, wo es viele mögliche Feldnamen gibt, der Knoten aber nur spärlich besetzt ist. Wenn man also nach 20 möglichen Feldern sucht, aber nur 3 überhaupt im Knoten vorkommen, wäre es deutlich schneller, den Knoten nur einmal zu "parsen" und dann zu schauen, welche Felder in der Suchliste vorkommen.

Das verwendete Pattern ist für wenige Feldnamen aber "große" Knoten besser. Oftmals dürfte der Aufrufer wissen, welche Variante wahrscheinlicher ist.